### ISSUE TITLE
[clang-format] Regression in current main branch (llvm 17)

### ISSUE THREAD

**Issue by Sedeniono** [llvm/llvm-project#62892]
Consider the following file `inputFile.hpp`:
```C++
struct foo {
	void test() {
#if 1
#else
#endif
	}
#if 1
private:


#endif
};
```
The new lines after the `private:` are important, as are the initial `#if 1..else..endif`. 

The `.clang-format` file contains a single line:
```yml
IndentPPDirectives: BeforeHash
```

Using the current LLVM main branch to format the whole file, this triggers the message
```console
The new replacement overlaps with an existing replacement.
New replacement: inputFile.hpp: 64:+3:"

"
Existing replacement: inputFile.hpp: 64:+3:"
"
```
while in Clang-Format 16 this does not happen.

I could bisect it to a4c87f8ccaccc76fd7d1c6c2e639ca84b9ec7794, which apparently is related to #61498. Differential revision: https://reviews.llvm.org/D150057
Maybe @owenca does have an idea?

I noticed the above regression while looking at the example from #57117, and managed to reduce it to the above

**Comment by owenca** [comment]
https://reviews.llvm.org/D151954
